src/create-report.ts

```diff
@@ -0,0 +1,44 @@
+import * as ls from './types';
+import * as api from './models';
+import { toSpan } from './transform';
+import { hexToDec } from './utils';
+import { ReadableSpan } from '@opentelemetry/tracing';
+
+export function createReport(
+  runtimeGUID: string,
+  accessToken: string,
+  reporterTags: { [key: string]: any }
+): (spans: ReadableSpan[]) => ls.ReportRequest {
+  const _runtimeGUID: string = runtimeGUID;
+  const _accessToken: string = accessToken;
+  const _auth: ls.Auth = new api.Auth({ accessToken: _accessToken });
+  const _reporterTags = reporterTags;
+  let _reporter: ls.Reporter;
+
+  function setReporter(
+    runtimeGUID: string,
+    attributes: { [key: string]: any }
+  ): ls.Reporter {
+    _reporter = new api.Reporter({
+      reporterId: hexToDec(runtimeGUID),
+      tags: Object.keys(attributes).map(
+        key => new api.KeyValue({ key: key, value: attributes[key] })
+      ),
+    });
+    return _reporter;
+  }
+
+  return function(spans: ReadableSpan[]): ls.ReportRequest {
+    return new api.ReportRequest({
+      auth: _auth,
+      reporter:
+        _reporter ||
```

Comment:
I was intentionally caching the reporter as I was under the impression that the resources would be static for the life of the process. I think this is technically true in OTel right now and it seems somewhat controversial to be passing the resource on a per-span basis. We can go with your suggestion though, as it will ensure we don't have a stale resource, if it does change.

Reply:
Based on the conversation at the JS SIG it seems like we probably can safely cache the reporter. Should we, or do you still prefer your suggestion. I've already updated the code for your suggestion, but it seems more efficient not to create the reporter for every post if you don't have to.

Reply:
I think it is fine then, maybe just leave message that if Resource are different within the export interval this should be refactored. Imho the resources attached to spans can suggests that they might be different, but this is something that otel spec should answer and probably change the spec then.